### PR TITLE
e2e: retry on 404 error for service reachability tests

### PR DESCRIPTION
**Pull request by yujuhong**

### CODE REVIEW

**Comment by bowei** [comment]
Can you add why we should ignore 404s?

**Comment by yujuhong** [comment]
> Can you add why we should ignore 404s?

I am not sure if we should. This is to work around the flakes I observed in the network tiers testing.

**Comment by MrHohn** [comment]
I think it is reasonable to tolerate the 404 case in network tiers test, but may be better to do that for network tier specifically?

Put this into `TestReachableHTTPWithContentTimeout()` may change the behavior for other network tests. Also it seems a bit odd that it returns an error when content is not as expected but returns nil when status code is unmatched.

**Comment by yujuhong** [comment]
@bowei @MrHohn, I updated the network tiers test to tolerate the 404 error. I've also tested this patch with my own cluster and haven't run into any failure after 5 runs.

**Comment by bowei** [comment]
thanks for the fix
/lgtm

File: test/e2e/framework/networking_utils.go [hunk 1]
```diff
@@ -679,6 +688,11 @@ func TestReachableHTTPWithContentTimeout(ip string, port int, request string, ex
 		return false, nil
 	}
 	if resp.StatusCode != 200 {
+		for _, code := range retriableErrCodes {
+			if resp.StatusCode == code {
+				return false, nil
```
Comment (MrHohn) [inline test/e2e/framework/networking_utils.go]: Would it help if we also log status code here?

**Review by MrHohn — APPROVED** [review]
Thanks! LGTM

**Comment by yujuhong** [comment]
The previous test failure was caused by kernel panic (known issue: #45368). Unrelated to this PR.
```
[  695.201205] BUG: unable to handle kernel NULL pointer dereference at 0000000000000078
[  695.209767] IP: [<ffffffff810a1130>] check_preempt_wakeup+0xd0/0x1d0
[  695.216462] PGD 1ac58e067 PUD 1efa08067 PMD 0 
[  695.221400] Oops: 0000 [#1] SMP 
[  695.224977] Modules linked in: nfsv3 rpcsec_gss_krb5 nfsv4 dns_resolver sg dummy xt_statistic nf_conntrack_netlink nfnetlink sch_htb ebt_ip ebtable_filter ebtables veth xt_nat xt_recent ipt_REJECT xt_mark xt_comment xt_tcpudp ipt_MASQUERADE iptable_filter iptable_nat nf_conntrack_ipv4 nf_defrag_ipv4 nf_nat_ipv4 xt_addrtype ip_tables xt_conntrack x_tables nf_nat nf_conntrack bridge stp llc aufs(C) nfsd auth_rpcgss oid_registry nfs_acl nfs lockd fscache sunrpc crct10dif_pclmul crc32_pclmul crc32c_intel aesni_intel aes_x86_64 parport_pc lrw gf128mul psmouse parport pvpanic glue_helper i2c_piix4 serio_raw i2c_core pcspkr processor button thermal_sys virtio_net ablk_helper cryptd evdev ext4 crc16 mbcache jbd2 sd_mod crc_t10dif crct10dif_common virtio_scsi scsi_mod virtio_pci virtio virtio_ring
[  695.305209] CPU: 1 PID: 8933 Comm: exe Tainted: G         C    3.16.0-4-amd64 #1 Debian 3.16.43-2+deb8u1
[  695.314800] Hardware name: Google Google Compute Engine/Google Compute Engine, BIOS Google 01/01/2011
[  695.324138] task: ffff8801ffe3d3f0 ti: ffff8801efbe8000 task.ti: ffff8801efbe8000
[  695.331732] RIP: 0010:[<ffffffff810a1130>]  [<ffffffff810a1130>] check_preempt_wakeup+0xd0/0x1d0
[  695.340757] RSP: 0018:ffff8801efbebe60  EFLAGS: 00010006
[  695.346187] RAX: 0000000000000001 RBX: ffff8801b459d700 RCX: 0000000000000008
[  695.353436] RDX: 0000000000000001 RSI: ffff8800bb55b3b0 RDI: ffff88021fc12fb8
[  695.360686] RBP: 0000000000000000 R08: ffffffff81610640 R09: 0000000000000001
[  695.367930] R10: 0000000000000001 R11: 0000000000000000 R12: ffff880073d80ca0
[  695.375180] R13: ffff88021fc12f40 R14: 0000000000000000 R15: 0000000000000000
[  695.382432] FS:  0000000001293880(0063) GS:ffff88021fd00000(0000) knlGS:0000000000000000
[  695.390633] CS:  0010 DS: 0000 ES: 0000 CR0: 0000000080050033
[  695.396487] CR2: 0000000000000078 CR3: 00000001efb13000 CR4: 00000000001406e0
[  695.403733] Stack:
[  695.405857]  0000000000012f40 ffff88021fc12f40 0000000000012f40 ffff88021fc12f40
[  695.413846]  ffff8800bb55ba34 0000000000000246 ffff8801ac606f40 ffffffff81095be5
[  695.421835]  ffff8800bb55b3b0 ffffffff810986ca 00007fffffffeffd 0000000000000000
[  695.429828] Call Trace:
[  695.432389]  [<ffffffff81095be5>] ? check_preempt_curr+0x85/0xa0
[  695.438771]  [<ffffffff810986ca>] ? wake_up_new_task+0xda/0x190
[  695.444988]  [<ffffffff81067a49>] ? do_fork+0x139/0x3d0
[  695.450323]  [<ffffffff8151a7f9>] ? stub_clone+0x69/0x90
[  695.455746]  [<ffffffff8151a48d>] ? system_call_fast_compare_end+0x10/0x15
[  695.462733] Code: 39 c2 7d 27 0f 1f 80 00 00 00 00 83 e8 01 48 8b 5b 70 39 d0 75 f5 48 8b 7d 78 48 3b 7b 78 74 15 0f 1f 00 48 8b 6d 70 48 8b 5b 70 <48> 8b 7d 78 48 3b 7b 78 75 ee 48 85 ff 74 e9 e8 8c cb ff ff 48 
[  695.489988] RIP  [<ffffffff810a1130>] check_preempt_wakeup+0xd0/0x1d0
[  695.496666]  RSP <ffff8801efbebe60>
[  695.500271] CR2: 0000000000000078
[  695.504201] ---[ end trace 9c0c0a6695662a67 ]---
[  695.508926] Kernel panic - not syncing: Fatal exception
[  696.576707] Shutting down cpus with NMI
[  696.581640] Kernel Offset: 0x0 from 0xffffffff81000000 (relocation range: 0xffffffff80000000-0xffffffff9fffffff)
[  696.591937] Rebooting in 10 seconds..
[  706.580807] ACPI MEMORY or I/O RESET_REG.
```

**Pull request merged**